src/Agent.Worker/JobRunner.cs

```diff
@@ -128,13 +128,21 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                         }
 
                         if (!string.IsNullOrWhiteSpace(notSupportNet6Message))
-                        {
+                        {                            
+                            if(!AgentKnobs.AcknowledgeNoUpdates.GetValue(jobContext).AsBoolean())
+                            {
+                                jobContext.AddIssue(new Issue(){Type = IssueType.Error,Message= StringUtil.Loc("FailAgentOnUnsupportedOs")});
```

Comment:
You can consider to use method `jobContext.Error(StringUtil.Loc("FailAgentOnUnsupportedOs"))`

Reply:
in this case, it ends with this error only. I guess there may be another error till this. But you a right, we just break the pipeline, it means there cannot be new error.  will check with your suggestion